### PR TITLE
feat/common_query_tests

### PR DESCRIPTION
**Pull request by NeonJarbas**
add initial unittests for common query and continuous dialog

also adds a new event to trigger common_query directly without going trough the intent service

### CODE REVIEW

File: test/unittests/common_query/test_common_query.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+        self.bus.emit(Message("common_query.question",
+                              {"utterance": "what is the speed of light"}))
+
+        expected = [
+            # original query
+            {'context': {},
+             'data': {'utterance': 'what is the speed of light'},
+             'type': 'common_query.question'},
+            # thinking animation
+            {'type': 'enclosure.mouth.think',
```
Comment (NeonDaniel) [inline test/unittests/common_query/test_common_query.py:37]: Should this be checked here? Seems like these messages belong more in a PHAL plugin?

File: test/unittests/common_query/test_continuous_dialog.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        def unset_context(message):
+            self.skill.has_context = False
+
+        self.bus.on('add_context', set_context)
+        self.bus.on('remove_context', unset_context)
+
+    def test_continuous_dialog(self):
+        self.bus.emitted_msgs = []
+
+        # "ask the wolf X"
```
Comment (NeonDaniel) [inline test/unittests/common_query/test_continuous_dialog.py:35]: wolfram or wiki?

File: mycroft/skills/intent_services/commonqa_service.py [hunk 3]
```diff
@@ -32,6 +32,7 @@ def __init__(self, bus):
         self.enclosure = EnclosureAPI(self.bus, self.skill_id)
         self._vocabs = {}
         self.bus.on('question:query.response', self.handle_query_response)
+        self.bus.on('common_query.question', self.handle_question)
```
Comment (NeonDaniel) [inline mycroft/skills/intent_services/commonqa_service.py:35]: This new bus handler should be noted in the PR description

**Review by NeonDaniel — COMMENTED** [review]
Looks good! left some minor feedback

File: test/unittests/common_query/test_common_query.py (hunk 1 above)
Comment (JarbasAl) [reply test/unittests/common_query/test_common_query.py:37]: some enclosure messages are emitted by core, PHAL plugins react to them if wanted, eg. mk1 faceplate or mk2 led animation

**Pull request merged**